src/transformers/modeling_tf_bart.py

```diff
@@ -0,0 +1,1183 @@
+# coding=utf-8
+# Copyright 2020 The Facebook AI Research Team Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TF BART model, ported from the fairseq repo."""
+import logging
+import math
+import random
+import warnings
+from typing import Dict, Optional, Tuple
+
+import tensorflow as tf
+from tensorflow import Tensor
+from tensorflow.keras.layers import Dense, LayerNormalization
+
+from .activations_tf import ACT2FN
+from .configuration_bart import BartConfig
+from .file_utils import add_start_docstrings, add_start_docstrings_to_callable
+from .modeling_tf_outputs import TFBaseModelOutput, TFBaseModelOutputWithPast, TFSeq2SeqLMOutput, TFSeq2SeqModelOutput
+
+# Public API
+from .modeling_tf_t5 import _NoLayerEmbedTokens
+from .modeling_tf_utils import (
+    DUMMY_INPUTS,
+    TFPreTrainedModel,
+    TFSharedEmbeddings,
+    cast_bool_to_primitive,
+    keras_serializable,
+    shape_list,
+)
+from .tokenization_utils_base import BatchEncoding
+
+
+def create_position_ids_from_input_ids(input_ids, padding_idx):
+    """Replace non-padding symbols with their position numbers. Position numbers begin at
+    padding_idx+1. Padding symbols are ignored. This is modified from fairseq's
+    `utils.make_positions`.
+
+    :param torch.Tensor x:
+    :return tf.Tensor:
+    """
+    mask = input_ids.ne(padding_idx).int()
+    incremental_indices = tf.cumsum(mask, axis=1).type_as(mask) * mask
+    return incremental_indices.long() + padding_idx
+
+
+def print_tensor(msg, t):  # DELEMETME
+    # assert t.shape
+    if t is None:
+        print(f"{msg}: {t}")
+        return
+    ndim = len(t.shape)
+    if ndim == 1:
+        slice = t[:3]
+    elif ndim == 2:
+        slice = t[:3, :3]
+    elif ndim == 3:
+        slice = t[:3, :3, :3]
+    elif ndim == 4:
+        slice = t[:3, :3, :3, :3]
+    print(f"{msg}: {slice}")
+
+
+logger = logging.getLogger(__name__)
+
+
+BART_START_DOCSTRING = r"""
```

Comment:
This should copy the `BERT_START_DOCSTRING` in `moveling_tf_bert.py` with BertConfig -> BartConfig.